### ISSUE TITLE
Align Tensor class tutorial with the Matlab version

### ISSUE THREAD

**Issue by DeepBlockDeepak** [sandialabs/pyttb#125]
The goal is to replicate the [Tensor Toolbox](https://tensortoolbox.org/tensor_doc.html) page within our Tensors.ipynb.

**Comment by DeepBlockDeepak** [comment]
## Initial Documentation Conversion Issues

I had a few conversion issues when trying to convert the [MATLAB tensor toolbox](https://tensortoolbox.org/tensor_doc.html) documentation into analogous Python documentation. Here are the remaining ones I haven't handled, with some examples of my attempts:

### 1. Subscripted Reference for a tensor

I'm unsure how to use the appropriate extraction.:

```Matlab
>>> subs = [[0,0,0,0], [2,3,4,0]]; 
>>> X[subs]

ans =

    0.0119
    0.9619
```

My initial attempt :
```python
>>> subs = [[0,0,0,0], [2,3,4,0]]
>>> X[subs] # Extract 2 values by subscript.
array([[0.5488135 , 0.5488135 , 0.5488135 , 0.5488135 ],
       [0.4236548 , 0.0202184 , 0.96366276, 0.5488135 ]])
```



### 2. Using end for the last array index
Currently I'm not sure how to replicate MATLAB's `end-1` style of functionality:
```Matlab
X(1,1,1:end-1)  %<-- Same as X(1,1,1:2).
ans is a tensor of size 2
	ans(:) = 
	     5
	     7
```
**And**
```Matlab
X(1,1,end+1) = 5 %<-- Same as X(1,1,4).
```

### 3. Use find for subscripts of nonzero elements of a tensor
Not sure how to use the 'extract' functionality. I couldn't replicate either of these in Python.

```Matlab
V = X(S) %<-- Extract the corresponding values from X.
V =

     2
     2
```
**AND**
```Matlab
X([1:6]','extract') %<-- Same thing *but* result is a vector.
```

### 4. Basic Operations
I'm not sure how to demonstrate rdivide explicitly without the operation automatically defaulting to performing identically to ldivide. Rdivide looks like it's supposed to be performed when the second operand doesn't have an available \__rdivide\__ but I got stuck trying to trigger that case.

### 5. Use permute to reorder the modes of a tensor
I'm not sure how to produce tensor that looks like a column vector as in the MATLAB example:

```Matlab
X = tensor(1:4,4); %<-- Create a 1-way tensor.
permute(X,1) %<-- Call permute with *only* one dimension.
ans is a tensor of size 4
	ans(:) = 
	     1
	     2
	     3
	     4
```

My attempt:
```python
>>> X = ttb.tensor().from_data(np.arange(1,5), (4,))
>>> X.permute(np.array(1,))
tensor of shape 4
data[:] = 
[1 2 3 4]
```

**Comment by dmdunla** [comment]
@DeepBlockDeepak can you create a PR (linked to your working branch) so that I can run your code to see if I can make some suggestions?

**Comment by ntjohnson1** [comment]
This is sort of tangential or a pre-req (but not necessarily a hard blocker). Have you two discussed plans for how we want to mange the jupyter notebooks? It might be nice to figure out this https://github.com/sandialabs/pyttb/issues/107 with the current content prior to merging the changes. It has been a little bit since I managed a repo with evolving notebooks but I recall them generating large binary diffs which rapidly bloat the repo and make it a bit more painful to work with.

**Comment by dmdunla** [comment]
Closed by 6b853d22fea49a2f39674e3816287901b04f1d89.

New issues tracked in #162
